compiler/luci/profile/src/CircleNodeOrigin.cpp

```diff
@@ -80,16 +80,13 @@ class CompositeOrigin final : public luci::CircleNodeOrigin
 public:
   CompositeOrigin() = delete;
 
-  template <typename T> CompositeOrigin(T origins)
+  template <typename T> CompositeOrigin(const T& origins)
   {
     if (origins.size() == 0)
       throw std::invalid_argument("No origins provided");
 
-    for (auto &origin : origins)
-    {
-      if (origin != nullptr)
-        _origins.emplace_back(origin);
-    }
+    std::copy_if(std::begin(origins), std::end(origins), std::back_inserter(_origins),
```

Comment:
plz add header for `copy_if`, `back_inserter`

Reply:
why is this change needed?

Reply:
This change just follows the approach which suggests using standard library algorithms over raw loops. It's a simple refactor of the existing code, not backed by any requirement.